### ISSUE TITLE
Add BLE transport

### ISSUE THREAD

**Issue by soundanalogous** [firmata/arduino#251]
The Arduino 101 has BLE. Add the ability to use BLE as a transport (so a user can communicate with the board using Firmata over BLE).

This will be challenging due to the limited size of data packets (20 bytes per packet and up to 6 packets per interval). Could use some help with this if there are any BLE experts who would like to see this functionality added to Firmata.

**Comment by soundanalogous** [comment]
Work in progress: https://github.com/firmata/arduino/tree/ble

**Comment by soundanalogous** [comment]
Further progress for Arduino 101 BLE is currently blocked by: https://github.com/01org/corelibs-arduino101/issues/95

**Comment by soundanalogous** [comment]
The work for this is mostly done. Will be merged into master once the [Arduino 101 core](https://github.com/01org/corelibs-arduino101) v1.0.5 update is released and updatable via the Arduino Boards Manager.

**Comment by soundanalogous** [comment]
Merged. However, `setConnectionInterval` is skipped for Arduino 101 by default. Should be available without patching when Intel releases the Curie Board v1.0.6 package. Until then CurieBLE can be patched which will result in a much better reporting interval for analog input and I2C continuous reads.
